src/Dialect/ONNX/ONNXShapeHelper.cpp

```diff
@@ -743,18 +743,20 @@ LogicalResult ONNXSplitOpShapeHelper::Compute(
     ONNXSplitOpAdaptor operandAdaptor) {
 
   auto split = op->split();
-  auto builder = mlir::Builder(op->getContext());
 
-  llvm::Optional<ArrayAttr> optionalAttr;
+  SmallVector<IndexExpr, 4> indexExprArray;
   if (auto splitConstOp = getONNXConstantOp(split)) {
-    // Checking value of split parameter.
-    auto splitAttribute = createArrayAttrFromConstantOp(builder, splitConstOp);
-    optionalAttr.emplace(splitAttribute);
+    Operation *genericOp = reinterpret_cast<Operation *>(op);
+    ArrayValueIndexCapture splitCapture(
+        genericOp, split, fGetDenseVal, fLoadVal);
+    auto splitRank =
+        splitConstOp.valueAttr().dyn_cast_or_null<DenseElementsAttr>().size();
+    splitCapture.getSymbolList(splitRank, indexExprArray);
   } else if (!split.getType().template isa<NoneType>()) {
     llvm_unreachable("dynamic split not yet supported");
```

Comment:
I think we can handle the dynamic split now.

Reply:
@chentong319 I think handling dynamic split might require either a separate set of patches or additional effort beyond the scope of this PR? If that's the case, should I keep the error message for now? Let me know what you think. Thanks

Reply:
Yes, we can have a separate PR.

Reply:
Great! Let me know if there are other suggestions. I currently don't have a timeline for implementing the dynamic split. Others can take over if they desire. There are be plans to do the same for Unsqueeze/Squeeze op to allow for BF16 support first.